### ISSUE TITLE
ConnectionRefusedError: [Errno 111] Connection refused

### ISSUE THREAD

**Issue by marticanyellest** [zecure/shadowd_python#21]
Hi, I have configured my main.py with the documentation configuration, but the flask server has stopped working (Internal Server Error). I have enabled debut and got the following output:

**Traceback (most recent call last):**
   File "/usr/local/lib/python3.10/site-packages/shadowd/connector.py", line 264, in start
     status = connection.send(
   File "/usr/local/lib/python3.10/site-packages/shadowd/connector.py", line 181, in send
     connection.connect((host, port))
ConnectionRefusedError: [Errno 111] Connection refused

**Config added in main.py:**
from shadowd.flask_connector import InputFlask, OutputFlask, Connector
app = Flask(__name__)
@app.before_request
def before_req():
    input = InputFlask(request)
    output = OutputFlask()
    Connector().start(input, output)

Any idea how to fix it?

Thank you very much.

**Comment by zit-hb** [comment]
Hello, the `ConnectionRefusedError: [Errno 111] Connection refused` seems like the host/port is wrong. What host have you specified in the config? Can you connect to it using netcat (`nc -v host 9115`)?

**Comment by marticanyellest** [comment]
Hello @zit-hb , the only port that I have changed, has been the one of the UI, the rest are the default ones (for the daemon, 127.0.0.1:9115). 
I tried connecting with netcat, and it worked.

**connectors.ini:**

[shadowd_python]
; Sets the profile id. Required.
profile=1

; Sets the key for the profile. Required.
key=test

; Sets the address of the shadowd server.
; Default Value: 127.0.0.1
;host=

; Sets the port of the shadowd server.
; Default Value: 9115
;port=

; Sets the path to the SSL certificate and enables SSL.
;ssl=

; If activated threats are not removed. This can be used to test new rules without
; making the web application unusable. It can be also used to turn Shadow Daemon
; into a high-interaction web honeypot.
; Possible Values:
;   0
;   1
; Default Value: 0
;observe=

; If activated error messages are printed.
; Possible Values:
;   0
;   1
; Default Value: 0
debug=1

; Sets the log file, but it is only used if debug is enabled.
; Default Value: /var/log/shadowd.log
log=/Users/Admin/Desktop/ShadowdLogs/shadowd.log

; If set the ignore list is used to ignore certain parameters and not send them to
; the shadowd server. It is good practise to not send passwords or other very
; sensitive information to the server.
;ignore=

; Sets the source for the client ip. It is a key of $_SERVER. If you are using a
; reverse proxy or something similar forward the real client ip and reference it
; here, e.g., HTTP_X_FORWARDED_FOR.
; Default Value:
;   CGI:    REMOTE_ADDR
;   Django: REMOTE_ADDR
;   Flask:  REMOTE_ADDR
;client_ip=

; Sets the source for the caller. It is a key of $_SERVER. If you are using web
; application with a front controller or something like that change it to PHP_SELF.
; Default Value:
;   CGI:    SCRIPT_FILENAME
;   Django: PATH_INFO
;   Flask:  PATH_INFO
;caller=

**Comment by zit-hb** [comment]
Is the application maybe running in a Docker container or anything like that?

**Comment by marticanyellest** [comment]
@zit-hb  yes! I am running all in docker with docker-compose. Is there a problem? I have ping from the app container, and it is able to reach the shadow daemon.

**Comment by zit-hb** [comment]
It is no problem to run it in Docker but usually you have to set `host` to the address of the host system, for example `host=192.168.0.10`.

**Comment by marticanyellest** [comment]
@zit-hb it finally works!

I explain what I did in case it helps someone...
As you said, the ip in the connectors.ini file had to be the docker internal one. The problem is that docker assigns an IP dynamically every time you start docker compose. So I've created a network and assigned them static ips like this:

networks:
       demo_network:
           ipv4_address: 172.27.0.6

Thanks a lot for the help!
